test/cluster_monitor.go

```
@@ -119,6 +119,31 @@ type ConditionsSummary struct {
+// maxConsecutiveMonitorFailures is the number of consecutive monitoring script failures
+// (where the management cluster is also unreachable) before aborting with a structured error.
+const maxConsecutiveMonitorFailures = 3
+
+// checkManagementClusterHealth runs a lightweight kubectl command against the management cluster
+// to determine if the cluster API server is reachable. Returns nil if healthy, or a descriptive
+// error with remediation guidance from DetectNetworkError if unreachable.
+func checkManagementClusterHealth(t *testing.T, kubeContext string) error {
+	t.Helper()
+
+	// #nosec G204 -- kubeContext is validated upstream as RFC 1123 compliant
+	cmd := exec.Command("kubectl", "--context", kubeContext, "get", "ns", "--request-timeout=5s")
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)